### DISCUSSION TITLE
Warning Connect to Remote SQL Instance

### DISCUSSION THREAD

**Discussion by blackbolt1** [dataplat/dbatools#8434] [Q&A] [answered]
Hi, can you help me with some tips. I'm trying to connect to a remote SQL Instance but a keep getting a warning for example:

WARNING: [16:52:20][Export-DbaLogin] Failure | The network path was not found

With local instance i dont have any problems.

**Comment by niphlod** [comment]
can you connect by any other means ? "network path not found" is usually when you can't open the socket to the destination, meaning either the resolved address is incorrect or there's a firewall in between.

**Reply by blackbolt1** [reply]
Yes, i can connect through SSMS, Azure Data Studio and SQLCMD.
I use the -debug parameter and got this:

DEBUG: 20913 | [08:11:58][Export-DbaLogin] Processing Server through InputObject
DEBUG: 4735 | [08:11:58][Connect-DbaInstance] Starting process block
DEBUG: 4746 | [08:11:58][Connect-DbaInstance] Immediately checking for Azure
DEBUG: 4634 | [08:11:58][Test-Azure] Test for Azure is negative
DEBUG: 4753 | [08:11:58][Connect-DbaInstance] Starting loop for 'XXXXXXXX.5000': ComputerName =
'XXXXXX.5000', InstanceName = 'MSSQLSERVER', IsLocalHost = 'False', Type = 'Default'
DEBUG: 4857 | [08:11:58][Connect-DbaInstance] String is passed in, will build server object from
instance object and other parameters, do some checks and then return the server object
DEBUG: 4634 | [08:11:58][Test-Azure] Test for Azure is negative
DEBUG: 4965 | [08:11:58][Connect-DbaInstance] authentication method is 'local sql'
DEBUG: 5004 | [08:11:58][Connect-DbaInstance] ApplicationName will be set to 'dbatools PowerShell
module - dbatools.io'
DEBUG: 5036 | [08:11:58][Connect-DbaInstance] ConnectionTimeout will be set to '15'
DEBUG: 5066 | [08:11:58][Connect-DbaInstance] PacketSize will be set to '4096'
DEBUG: 5085 | [08:11:58][Connect-DbaInstance] Pooled will be set to 'True'
DEBUG: 5094 | [08:11:58][Connect-DbaInstance] SecurePassword will be set
DEBUG: 5120 | [08:11:58][Connect-DbaInstance] UserName will be set to 'test'
DEBUG: 5154 | [08:11:58][Connect-DbaInstance] Building ServerConnection from SqlConnectionInfo
DEBUG: 5156 | [08:11:58][Connect-DbaInstance] ServerConnection was built
DEBUG: 5174 | [08:11:58][Connect-DbaInstance] Building Server from ServerConnection
DEBUG: 5176 | [08:11:58][Connect-DbaInstance] Server was built
DEBUG: 5195 | [08:11:58][Connect-DbaInstance] Setting ConnectionContext.StatementTimeout to '0'
DEBUG: 5200 | [08:11:58][Connect-DbaInstance] The masked server.ConnectionContext.ConnectionString
is Data Source=XXXXXXX.5000;User ID=test;Password=********;Multiple Active Result
Sets=False;Encrypt=False;Trust Server Certificate=False;Packet Size=4096;Application Name="dbatools
 PowerShell module - dbatools.io"
DEBUG: 5210 | [08:11:58][Connect-DbaInstance] We connect to the instance by running SELECT
'dbatools is opening a new connection'
DEBUG: [08:12:13][Connect-DbaInstance] Failure | Error connecting to [XXXXXXXX.5000]: The network
path was not found

**Reply by niphlod** [reply]
**.5000** is a rather strange domain, are you sure your hostname is xxxxx.5000 and not, for example , xxxxx running on port 5000 (in which case, the notation should be xxxxx:5000 or xxxxx,5000) ?

**Reply by blackbolt1** [reply]
Yeah, you're right the problem was the port, it works with XXXXX:5000, using " , " wont work.

Thanks!

**Reply by niphlod** [reply]
perfect, glad to be of help. maybe set the question as answered if you don't have any other issue

**Reply by blackbolt1** [reply]
Now i have a different issue with the command Export-DbaLinkedServer.

Export-DbaLinkedServer -SqlInstance XXXXXXX:5000 -SqlCredential $wincred.

DEBUG: 4735 | [10:21:53][Connect-DbaInstance] Starting process block
DEBUG: 4746 | [10:21:53][Connect-DbaInstance] Immediately checking for Azure
DEBUG: 4634 | [10:21:53][Test-Azure] Test for Azure is negative
DEBUG: 4753 | [10:21:53][Connect-DbaInstance] Starting loop for 'XXXXXXXXX,PORT': ComputerName = 'XXXXXXXXX',
InstanceName = 'MSSQLSERVER', IsLocalHost = 'False', Type = 'Default'
DEBUG: 4857 | [10:21:53][Connect-DbaInstance] String is passed in, will build server object from instance object and
other parameters, do some checks and then return the server object
DEBUG: 4634 | [10:21:53][Test-Azure] Test for Azure is negative
DEBUG: 4965 | [10:21:53][Connect-DbaInstance] authentication method is 'local ad'
DEBUG: 5004 | [10:21:53][Connect-DbaInstance] ApplicationName will be set to 'dbatools PowerShell module - dbatools.io'
DEBUG: 5036 | [10:21:53][Connect-DbaInstance] ConnectionTimeout will be set to '15'
DEBUG: 5066 | [10:21:53][Connect-DbaInstance] PacketSize will be set to '4096'
DEBUG: 5085 | [10:21:53][Connect-DbaInstance] Pooled will be set to 'True'
DEBUG: 5154 | [10:21:53][Connect-DbaInstance] Building ServerConnection from SqlConnectionInfo
DEBUG: 5156 | [10:21:53][Connect-DbaInstance] ServerConnection was built
DEBUG: 5164 | [10:21:53][Connect-DbaInstance] ConnectAsUser will be set to 'True'
DEBUG: 5167 | [10:21:53][Connect-DbaInstance] ConnectAsUserName will be set to 'sqlinfra@xxxmty'
DEBUG: 5170 | [10:21:53][Connect-DbaInstance] ConnectAsUserPassword will be set
DEBUG: 5174 | [10:21:53][Connect-DbaInstance] Building Server from ServerConnection
DEBUG: 5176 | [10:21:53][Connect-DbaInstance] Server was built
DEBUG: 5195 | [10:21:53][Connect-DbaInstance] Setting ConnectionContext.StatementTimeout to '0'
DEBUG: 5200 | [10:21:53][Connect-DbaInstance] The masked server.ConnectionContext.ConnectionString is Data
Source=XXXXXXXXX,PORT;Integrated Security=True;Multiple Active Result Sets=False;Encrypt=False;Trust Server
Certificate=False;Packet Size=4096;Application Name="dbatools PowerShell module - dbatools.io"
DEBUG: 5210 | [10:21:53][Connect-DbaInstance] We connect to the instance by running SELECT 'dbatools is opening a new
connection'
DEBUG: 5219 | [10:21:54][Connect-DbaInstance] We have a connected server object
DEBUG: 5265 | [10:21:56][Connect-DbaInstance] We will set server.ComputerName to server.NetName
DEBUG: 5271 | [10:21:56][Connect-DbaInstance] ComputerName will be set to XXXXXXXXX
DEBUG: 4634 | [10:21:56][Test-Azure] Test for Azure is negative
DEBUG: 5281 | [10:21:56][Connect-DbaInstance] We added IsAzure = 'False', DbaInstanceName = instance.InstanceName =
'MSSQLSERVER', SqlInstance = server.DomainInstanceName = '', NetPort = instance.Port = 'PORT', ConnectedAs =
server.ConnectionContext.TrueLogin = 'xxxMTY\sqlinfra'
DEBUG: 5284 | [10:21:56][Connect-DbaInstance] We return the server object
DEBUG: 5312 | [10:21:56][Connect-DbaInstance] SetDefaultInitFields will be used
DEBUG: 136465 | [10:21:56][Add-ConnectionHashValue] Adding to connection hash
DEBUG: 5341 | [10:21:56][Connect-DbaInstance] We are finished with this instance
DEBUG: 4735 | [10:21:57][Connect-DbaInstance] Starting process block
DEBUG: 4746 | [10:21:57][Connect-DbaInstance] Immediately checking for Azure
DEBUG: 4634 | [10:21:57][Test-Azure] Test for Azure is negative
DEBUG: 4753 | [10:21:57][Connect-DbaInstance] Starting loop for 'XXXXXXXXX,PORT': ComputerName = 'XXXXXXXXX',
InstanceName = 'MSSQLSERVER', IsLocalHost = 'False', Type = 'Default'
DEBUG: 4857 | [10:21:57][Connect-DbaInstance] String is passed in, will build server object from instance object and
other parameters, do some checks and then return the server object
DEBUG: 4634 | [10:21:57][Test-Azure] Test for Azure is negative
DEBUG: 4965 | [10:21:57][Connect-DbaInstance] authentication method is 'local ad'
DEBUG: 5004 | [10:21:57][Connect-DbaInstance] ApplicationName will be set to 'dbatools PowerShell module - dbatools.io'
DEBUG: 5036 | [10:21:57][Connect-DbaInstance] ConnectionTimeout will be set to '15'
DEBUG: 5066 | [10:21:57][Connect-DbaInstance] PacketSize will be set to '4096'
DEBUG: 5085 | [10:21:57][Connect-DbaInstance] Pooled will be set to 'True'
DEBUG: 5154 | [10:21:57][Connect-DbaInstance] Building ServerConnection from SqlConnectionInfo
DEBUG: 5156 | [10:21:57][Connect-DbaInstance] ServerConnection was built
DEBUG: 5164 | [10:21:57][Connect-DbaInstance] ConnectAsUser will be set to 'True'
DEBUG: 5167 | [10:21:57][Connect-DbaInstance] ConnectAsUserName will be set to 'sqlinfra@xxxmty'
DEBUG: 5170 | [10:21:57][Connect-DbaInstance] ConnectAsUserPassword will be set
DEBUG: 5174 | [10:21:57][Connect-DbaInstance] Building Server from ServerConnection
DEBUG: 5176 | [10:21:57][Connect-DbaInstance] Server was built
DEBUG: 5195 | [10:21:57][Connect-DbaInstance] Setting ConnectionContext.StatementTimeout to '0'
DEBUG: 5200 | [10:21:57][Connect-DbaInstance] The masked server.ConnectionContext.ConnectionString is Data
Source=XXXXXXXXX,PORT;Integrated Security=True;Multiple Active Result Sets=False;Encrypt=False;Trust Server
Certificate=False;Packet Size=4096;Application Name="dbatools PowerShell module - dbatools.io"
DEBUG: 5210 | [10:21:57][Connect-DbaInstance] We connect to the instance by running SELECT 'dbatools is opening a new
connection'
DEBUG: 5219 | [10:21:57][Connect-DbaInstance] We have a connected server object
DEBUG: 5265 | [10:21:59][Connect-DbaInstance] We will set server.ComputerName to server.NetName
DEBUG: 5271 | [10:21:59][Connect-DbaInstance] ComputerName will be set to XXXXXXXXX
DEBUG: 4634 | [10:21:59][Test-Azure] Test for Azure is negative
DEBUG: 5281 | [10:21:59][Connect-DbaInstance] We added IsAzure = 'False', DbaInstanceName = instance.InstanceName =
'MSSQLSERVER', SqlInstance = server.DomainInstanceName = '', NetPort = instance.Port = 'PORT', ConnectedAs =
server.ConnectionContext.TrueLogin = 'xxxMTY\sqlinfra'
DEBUG: 5284 | [10:21:59][Connect-DbaInstance] We return the server object
DEBUG: 5312 | [10:22:00][Connect-DbaInstance] SetDefaultInitFields will be used
DEBUG: 136465 | [10:22:00][Add-ConnectionHashValue] Adding to connection hash
DEBUG: 5341 | [10:22:00][Connect-DbaInstance] We are finished with this instance
DEBUG: 20641 | [10:22:00][Export-DbaLinkedServer] Getting FullComputerName name for XXXXXXXXX,PORT.
DEBUG: 107594 | [10:22:00][Resolve-DbaNetworkName] Resolving XXXXXXXXX using .NET.Dns GetHostEntry
DEBUG: 107604 | [10:22:00][Resolve-DbaNetworkName] Resolving 192.168.32.50 using .NET.Dns GetHostByAddress
DEBUG: 107674 | [10:22:00][Resolve-DbaNetworkName] Getting domain name from the remote host XXXXXXXXX.ad.xxx.com
DEBUG: 32707 | [10:22:00][Get-DbaCmObject] Configuration loaded | Cache disabled: False
DEBUG: 32720 | [10:22:00][Get-DbaCmObject] [XXXXXXXXX] Retrieving Management Information
DEBUG: 32760 | [10:22:00][Get-DbaCmObject] [XXXXXXXXX] Accessing computer using Cim over WinRM
DEBUG: 32765 | [10:22:00][Get-DbaCmObject] [XXXXXXXXX] Accessing computer using Cim over WinRM - Success
DEBUG: 107695 | [10:22:00][Resolve-DbaNetworkName] Getting DNS domain from the remote host XXXXXXXXX
DEBUG: 141917 | [10:22:00][Invoke-Command2] Found an existing session
dbatools_Default_16f716ce-6b90-4160-9ca1-e33155cc6cd2_XXXXXXXXX, reusing it
DEBUG: 107713 | [10:22:00][Resolve-DbaNetworkName] Resolving XXXXXXXXX.ad.xxx.com using .NET.Dns GetHostEntry
DEBUG: 144472 | [10:22:00][Resolve-DbaComputerName] XXXXXXXXX,PORT is not the the local host, so we use
FullComputerName.
DEBUG: 144475 | [10:22:00][Resolve-DbaComputerName] Resolved 'XXXXXXXXX,PORT' to 'XXXXXXXXX.ad.xxx.com'
DEBUG: 20644 | [10:22:00][Export-DbaLinkedServer] Checking if Remote Registry is enabled on XXXXXXXXX,PORT.
DEBUG: 141917 | [10:22:00][Invoke-Command2] Found an existing session
dbatools_Default_16f716ce-6b90-4160-9ca1-e33155cc6cd2_XXXXXXXXX.ad.xxx.com, reusing it
DEBUG: 139003 | [10:22:00][Get-DecryptedObject] Querying service master key
DEBUG: 107594 | [10:22:01][Resolve-DbaNetworkName] Resolving XXXXXXXXX using .NET.Dns GetHostEntry
DEBUG: 107604 | [10:22:01][Resolve-DbaNetworkName] Resolving 192.168.32.50 using .NET.Dns GetHostByAddress
DEBUG: 107674 | [10:22:01][Resolve-DbaNetworkName] Getting domain name from the remote host XXXXXXXXX.ad.xxx.com
DEBUG: 32707 | [10:22:01][Get-DbaCmObject] Configuration loaded | Cache disabled: False
DEBUG: 32720 | [10:22:01][Get-DbaCmObject] [XXXXXXXXX] Retrieving Management Information
DEBUG: 32760 | [10:22:01][Get-DbaCmObject] [XXXXXXXXX] Accessing computer using Cim over WinRM
DEBUG: 32765 | [10:22:01][Get-DbaCmObject] [XXXXXXXXX] Accessing computer using Cim over WinRM - Success
DEBUG: 107695 | [10:22:01][Resolve-DbaNetworkName] Getting DNS domain from the remote host XXXXXXXXX
DEBUG: 141917 | [10:22:01][Invoke-Command2] Found an existing session
dbatools_Default_16f716ce-6b90-4160-9ca1-e33155cc6cd2_XXXXXXXXX, reusing it
DEBUG: 107713 | [10:22:01][Resolve-DbaNetworkName] Resolving XXXXXXXXX.ad.xxx.com using .NET.Dns GetHostEntry
DEBUG: 144472 | [10:22:01][Resolve-DbaComputerName] XXXXXXXXX,PORT is not the the local host, so we use
FullComputerName.
DEBUG: 144475 | [10:22:01][Resolve-DbaComputerName] Resolved 'XXXXXXXXX,PORT' to 'XXXXXXXXX.ad.xxx.com'
DEBUG: 139019 | [10:22:02][Get-DecryptedObject] Decrypt the service master key
DEBUG: 141917 | [10:22:02][Invoke-Command2] Found an existing session
dbatools_Default_16f716ce-6b90-4160-9ca1-e33155cc6cd2_XXXXXXXXX.ad.xxx.com, reusing it
Exception calling "Unprotect" with "3" argument(s): "Key not valid for use in specified state.
"
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : CryptographicException
    + PSComputerName        : XXXXXXXXX.ad.xxx.com

DEBUG: 139037 | [10:22:02][Get-DecryptedObject] Choose the encryption algorithm based on the SMK length - 3DES for
2008, AES for 2012
DEBUG: 139040 | [10:22:02][Get-DecryptedObject] ServiceKey found: .Length
WARNING: [10:22:02][Get-DecryptedObject] Unknown key size. Do you have administrative access to the Windows registry on
  Otherwise, we're out of ideas.
DEBUG: [10:22:02][Get-DecryptedObject] Unknown key size. Do you have administrative access to the Windows registry on
Otherwise, we're out of ideas.

**Reply by niphlod** [reply]
aaaand, as stated, are you sure you have administrative access to the windows registry ?

**Reply by blackbolt1** [reply]
I know the user I'm using is sysadmin on the instance, but I don't know if it's sysadmin on the machine.

It has to?

**Reply by niphlod** [reply]
yes, because SQL stores linked server credentials encrypted with a key which is stored on the registry of the server

**Comment by blackbolt1** [comment, marked as answer]
Thanks for the help!
